src/main/java/uk/gov/companieshouse/acsp/manage/users/service/AcspMembersService.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.companieshouse.acsp.manage.users.service;
+
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.acsp.manage.users.model.AcspMembersDao;
+import uk.gov.companieshouse.acsp.manage.users.repositories.AcspMembersRepository;
+import uk.gov.companieshouse.api.acsp_manage_users.model.AcspMembership;
+
+import java.time.LocalDateTime;
+import java.util.Objects;
+
+import static uk.gov.companieshouse.GenerateEtagUtil.generateEtag;
+
+@Service
+public class AcspMembersService {
+    private final AcspMembersRepository acspMembersRepository;
+
+    public AcspMembersService(AcspMembersRepository acspMembersRepository) {
+        this.acspMembersRepository = acspMembersRepository;
+    }
+
+    public AcspMembersDao createAcspMembersWithOwnerRole(String acspNumber, String userId) {
+        if (Objects.isNull(acspNumber)) {
+            throw new NullPointerException("AcspNumber should be provided");
+        }
+
+        if (Objects.isNull(userId)) {
+            throw new NullPointerException("UserId should be provided");
+        }
```

Comment:
Subjective thing.  But I'm not a fan of the null checks, and then throwing the same error anyways, I prefer these checks to be done outside of the calling method by whoever is responsible for calling them - I feel they pollute the method.  But again, subjective and opinion based, think the code is fine

Reply:
I'm probably a bit paranoid ;-)